### PR TITLE
[Demian Moschin]-0.1-Ready for revision

### PR DESCRIPTION
**Pull request by Demian-Moschin**
@troianoandres 
@Tergopol 

**Work done:**
- **Ex0.1.1:** done.
- **Ex0.1.2:** done.
- **Ex0.1.3:** done.
- **Ex0.2.1:** done.

**Comments:**
Still not able to merge, work in progress.

### CODE REVIEW

File: 00-HTML-CSS-basics/HTML/Excersise1/media.html [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+
+  <body>
+    
+
+    <section>
+      <header>
+        <h1>Videos</h1>
+      </header>
+      <video controls autoplay loop muted>
+        <source src="media/jquery.webm" type="video/webm" />
```
Comment (troyax) [inline 00-HTML-CSS-basics/HTML/Excersise1/media.html:29]: You'll need to add some other audio & video formats, to keep browser compatibility. The link provided into the repo it's enought information I think.

File: 00-HTML-CSS-basics/HTML/Excersise1/tableOfExpenses.html [hunk 2]
```diff
@@ -0,0 +1,106 @@
+<!DOCTYPE html>
```
Comment (troyax) [inline 00-HTML-CSS-basics/HTML/Excersise1/tableOfExpenses.html]: The file name it's OK, but as a good practice I think it's better to use `table-of-expenses.html` as example

Comment (troyax) [inline 00-HTML-CSS-basics/HTML/Excersise1/tableOfExpenses.html]: The table it's OK, but the requirement was to fill it with expenses. Anyway well done :)

File: 00-HTML-CSS-basics/HTML/Excersise2/form.html [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+
+  </head>
+
+  <body>
+
+    <form id="sign-form" accept-charset="UTF-8">
+
+      <fieldset>
+        <legend>Personal information: </legend>
+        <p>
```
Comment (troyax) [inline 00-HTML-CSS-basics/HTML/Excersise2/form.html]: The paragraphs are not needed.

File: 00-HTML-CSS-basics/HTML/Excersise2/form.html [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+         <label>First Name:<input form="sign-form" autofocus name="firstName" required size="20" maxlength="20" placeholder="e.g.: John" /></label>
```
Comment (troyax) [inline 00-HTML-CSS-basics/HTML/Excersise2/form.html]: The form attribute is not suported by all the browsers, I think it's better to wrap all the input fields inside the form tag. Also the name attribute and ids attributes are better defined like `first-name` insstead of `firstName`.

File: 00-HTML-CSS-basics/HTML/Excersise2/form.html [hunk 5]
```diff
@@ -0,0 +1,64 @@
...
+        <p>
+          <label>E-Mail:<input type="email"  name="email" placeholder="user@mydomain.com" form="sign-form" required pattern=".*@*$.*"/></label>
+        </p>
+      </fieldset>
+      
+      <fieldset>
+        <legend>Tell us a little bit more about you!</legend>
+        <p>
+          <label>Choose your favourite sport:
+            <select>
```
Comment (troianoandres) [inline 00-HTML-CSS-basics/HTML/Excersise2/form.html]: This is not related to any form, try to wrap it with all the other inputs.

File: 00-HTML-CSS-basics/HTML/Excersise2/form.html (hunk 4 above)
Comment (troyax) [inline 00-HTML-CSS-basics/HTML/Excersise2/form.html]: Sorry my bad, I have not seen the form closing tag. Remove the form attribute from the inputs and it's just fine. :)

**Comment by Demian-Moschin** [comment]
@troianoandres 
Last issues fixed and ready for revision.

When validating the file "form.html" in w3c, the report tells me the "date" input is not valid in all browsers. 
Do I have an alternative, in order to put another date input tag?
Thanks

**Comment by troyax** [comment]
@Demian-Moschin actually this is known, usually you should use a custom "datepicker" to fix this issue. Anyway for this exercise a input with date type it's fine. Good work!

**Comment by Demian-Moschin** [comment]
@troianoandres 
Just pushed until exercise 2.3 of CSS -lab. Waiting for comments!

File: 00-HTML-CSS-basics/CSS/Basics/css/style.css [hunk 6]
```diff
@@ -0,0 +1,52 @@
...
+  src:url('../fonts/amatic/Amatic-Bold.ttf');
+  font-weight: bold;
+  font-style: normal;
+  font-size: 14px;
+ 
+}
+
+/*element selectors*/
+/*----------------*/
+html{
```
Comment (troyax) [inline 00-HTML-CSS-basics/CSS/Basics/css/style.css]: **Minor:** As a css styling we should leave 1 white space between the selector name and the curly brace after it.

File: 00-HTML-CSS-basics/CSS/Basics/css/style.css [hunk 7]
```diff
@@ -0,0 +1,52 @@
...
+  font-family: MyFont, Ubuntu, Arial, TimesNewRoman, sans-serif;
+}
+
+header,footer{
```
Comment (troyax) [inline 00-HTML-CSS-basics/CSS/Basics/css/style.css]: **Minor:** Also when grouping we should write one selector per line, like this: 

```
header,
footer {
  ...
}
```

File: 00-HTML-CSS-basics/CSS/Basics/css/style.css [hunk 8]
```diff
@@ -0,0 +1,52 @@
...
+/*Class selectors*/
+/*---------------*/
+
+/*class attribute that ends with 'r'* (already validated) */ 
+*[class$="r"]{
+  background-color: magenta;
+}
+
+/*class attribute containing 'a', which doesn't end with 'r' (already validated) */
+*[class*="a"] *:not([class$="r"]) {
```
Comment (troyax) [inline 00-HTML-CSS-basics/CSS/Basics/css/style.css]: I think this is not quite right, because this selector will match all classes that have a and also the ones without r, but you should combine both.

File: 00-HTML-CSS-basics/CSS/Basics/css/style.css (hunk 7 above)
Comment (troyax) [inline 00-HTML-CSS-basics/CSS/Basics/css/style.css]: Also why you use html tags as selectors intead of classes? As a good practice we should use in most cases just classes.

File: 00-HTML-CSS-basics/CSS/Basics/index.html [hunk 9]
```diff
@@ -0,0 +1,62 @@
+<!doctype html>
+<html>
+    <head>
+        <meta charset="utf-8">
+        <meta http-equiv="x-ua-compatible" content="ie=edge">
+        
+        <meta name="description" content="">
```
Comment (troyax) [inline 00-HTML-CSS-basics/CSS/Basics/index.html]: Take care of self closing tags, try to fix this in the future :)

File: 00-HTML-CSS-basics/HTML/Excersise1/helloWorld.html [hunk 10]
```diff
@@ -0,0 +1,36 @@
+<!DOCTYPE html>
```
Comment (troyax) [inline 00-HTML-CSS-basics/HTML/Excersise1/helloWorld.html]: We should try to use the following naming convention for files, this can cause some issues in the future. As example this file should be called:

`hello-world.html` instead of `helloWorld.html`

**Comment by troyax** [comment]
Review done, waiting for grid and media exercise.

File: 00-HTML-CSS-basics/CSS/Layout 2.5.1/css/style.css [hunk 11]
```diff
@@ -0,0 +1,113 @@
...
+  display: inline-block;
+}
+
+/*class attribute that ends with 'r'* (already validated) */ 
+*[class$="r"] {
+  background-color: magenta;
+}
+
+/*class attribute containing 'a', which doesn't end with 'r' (already validated) */
+*[class*="a"]*:not([class$="r"]) {
```
Comment (troyax) [inline 00-HTML-CSS-basics/CSS/Layout 2.5.1/css/style.css]: - are really needed?

File: 00-HTML-CSS-basics/CSS/Layout 2.5.1/css/style.css [hunk 12]
```diff
@@ -0,0 +1,113 @@
...
+  background-color: blue;
+}
+
+
+/*Media queries*/
+/*------------*/
+
+@media screen and (max-width:599px) {
+  *[class] {
```
Comment (troyax) [inline 00-HTML-CSS-basics/CSS/Layout 2.5.1/css/style.css]: why the [class] ?

File: 00-HTML-CSS-basics/CSS/Layout 2.5.1/css/style.css [hunk 13]
```diff
@@ -0,0 +1,113 @@
...
+  }
+
+  
+}
+
+@media screen and (min-width:600px) {
+  .navigation {
+      float: left;
+      width: 30%;
+      height: 100%;
```
Comment (troyax) [inline 00-HTML-CSS-basics/CSS/Layout 2.5.1/css/style.css]: you should be careful with the height: 100%, this may cause some trobules on some cases. If there's no problem for this one it's fine

File: 00-HTML-CSS-basics/CSS/Layout 2.5.1/css/style.css [hunk 14]
```diff
@@ -0,0 +1,113 @@
...
+      font-size: 1rem; /*14px*/
+    }
+
+  .sidebar {
+      float: right;
+      width: 30%;
+      height: 100%;
+      font-size: 0.714rem; /*10px*/
+    }
```
Comment (troyax) [inline 00-HTML-CSS-basics/CSS/Layout 2.5.1/css/style.css]: add line here

File: 00-HTML-CSS-basics/CSS/Layout 2.5.2/css/style.css [hunk 15]
```diff
@@ -0,0 +1,133 @@
...
+.clearfix:after {
+  clear: both;
+}
+
+.clearfix {
+  *zoom: 1;
+}
+
+
+.odd-section {
```
Comment (troyax) [inline 00-HTML-CSS-basics/CSS/Layout 2.5.2/css/style.css]: From here to line 100 why is needed?

File: 00-HTML-CSS-basics/CSS/Layout 2.5.2/css/style.css [hunk 16]
```diff
@@ -0,0 +1,133 @@
...
+  .column.full { width: 100%; }
+  .column.two-thirds { width: 66.7%; }
+  .column.half { width: 50%; }
+  .column.third { width: 33.3%; }
+  .column.fourth { width: 25%; }
+  .column.flow-opposite { float: right; }
+
+}
+
+@media (min-width: 40rem) {
```
Comment (troyax) [inline 00-HTML-CSS-basics/CSS/Layout 2.5.2/css/style.css]: Why not one media query only? What about other media queries (for different devices)

File: 00-HTML-CSS-basics/CSS/Layout 2.5.1/css/style.css (hunk 11 above)
Comment (Demian-Moschin) [inline 00-HTML-CSS-basics/CSS/Layout 2.5.1/css/style.css]: Not needed. It was left from the previous excersise! I will remove it in the next commit.

File: 00-HTML-CSS-basics/CSS/Layout 2.5.2/css/style.css (hunk 15 above)
Comment (Demian-Moschin) [inline 00-HTML-CSS-basics/CSS/Layout 2.5.2/css/style.css]: @troianoandres 
This is the structure defined in http://adamkaplan.me/grid/ (step 7)

the .clearfix is defined in order to have compatibility with IE 6/7 browsers
